tests/e2e/test_service.go

```diff
@@ -102,23 +102,19 @@ func (a *alertStore) GetAll() []string {
 }
 
 type snapStore struct {
-	sync.Mutex
-	d map[uint64]*protocol.SignedSnapshot
+	d *sync.Map
 }
 
 func (s *snapStore) Put(b *protocol.BatchSnapshots) {
-	s.Lock()
-	defer s.Unlock()
-
 	for _, snap := range b.Snapshots {
-		s.d[snap.Snapshot.Version] = snap
+		s.d.Store(snap.Snapshot.Version, snap)
 	}
 }
 
 func (s *snapStore) Get(version uint64) (v *protocol.SignedSnapshot, ok bool) {
-	s.Lock()
-	defer s.Unlock()
-	v, ok = s.d[version]
+	var tmpV interface{}
+	tmpV, ok = s.d.Load(version)
+	v = tmpV.(*protocol.SignedSnapshot)
```

Comment:
Can't you summarize it this way?

```
tmpV, ok = s.d.Load(version)
return tmpV.(*protocol.SignedSnapshot), ok
```

Reply:
For that to work I would have to change the signature of the function, which currently reads:
`func (s *snapStore) Get(version uint64) (v *protocol.SignedSnapshot, ok bool) {`
That forces me to assign to `v` the return value.

I reckon that @gdiazlo wrote the signature like that to enhance readability.  Thoughts?

Reply:
The compiler will optimize away the tmpV. This whole service will go to hell before ml ends, so I don't really care that much about its code style.

Reply:
Ok, then.  I'll leave it as it is :-)